sdk/keyvault/keyvault-secrets/test/public/utils/testAuthentication.ts

```diff
@@ -2,47 +2,73 @@
 // Licensed under the MIT license.
 
 import { SecretClient } from "../../../src";
-import { RecorderEnvironmentSetup, env, record } from "@azure-tools/test-recorder";
+import {
+  assertEnvironmentVariable,
+  env,
+  isRecordMode,
+  Recorder,
+  RecorderStartOptions,
+} from "@azure-tools/test-recorder";
 import { uniqueString } from "./recorderUtils";
 import TestClient from "./testClient";
 import { Context } from "mocha";
 import { getServiceVersion } from "./common";
-import { createXhrHttpClient, isNode } from "@azure/test-utils";
 import { createTestCredential } from "@azure-tools/test-credential";
+import { FindReplaceSanitizer } from "@azure-tools/test-recorder/types/src/utils/utils";
 
 export async function authenticate(
   that: Context,
   serviceVersion: ReturnType<typeof getServiceVersion>
 ): Promise<any> {
+  const generalSanitizers: FindReplaceSanitizer[] = [];
+
   const secretSuffix = uniqueString();
-  const recorderEnvSetup: RecorderEnvironmentSetup = {
-    replaceableVariables: {
+  if (isRecordMode()) {
+    generalSanitizers.push({
+      target: secretSuffix,
+      value: "",
+    });
+
+    // Add logic to sanitize the hostname of the url.
+    // Some nextLink URLs from paging include the port number, e.g. https://myvault.vault.azure.net:443/... this means that
```

Comment:
interesting, is this the first time we are seeing such?

Reply:
Yeah, I haven't seen this before, even with the other Key Vault packages that have been migrated. It seems like an unusual thing for the service to do, even if it is technically valid.